### PR TITLE
Include path to MSBuild in node handshake salt

### LINKED ISSUE #3137: Avoid crosstalk between same-MSBuild-but-different-SDKs nodes
~We should add the environment variables `DOTNET_HOST_PATH` and `DOTNET_ROOT` to the node-compatibility handshake computation.~

### Steps to reproduce

* Have two copies of the .NET Core SDK that use the same version of MSBuild but that aren't identical
* Run a build with copy A (created nodes will wait for reuse after the build completes)
* Run a build with copy B

### Expected  behavior

The build in copy B's dotnet host does not reuse nodes from copy A

### Actual behavior

This will be mostly true now, because the handshake includes `AssemblyTimestamp` which is `LastWriteTime` which will _probably_ be distinct, but isn't necessarily.

### PR DESCRIPTION
**Pull request by Forgind**
Prevent unwanted node reuse.

Fixes #3137.

### CODE REVIEW

File: src/Shared/CommunicationsUtilities.cs [hunk 1]
```diff
@@ -655,7 +655,9 @@ internal static void Trace(int nodeId, string format, params object[] args)
         /// <returns>Base Handshake</returns>
         private static long GetBaseHandshakeForContext(TaskHostContext hostContext)
         {
-            string salt = Environment.GetEnvironmentVariable("MSBUILDNODEHANDSHAKESALT");
+            string salt = Environment.GetEnvironmentVariable("MSBUILDNODEHANDSHAKESALT") +
+                Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") +
+                Environment.GetEnvironmentVariable("DOTNET_ROOT");
```
Comment (rainersigwald) [inline src/Shared/CommunicationsUtilities.cs]: Should we be using environment variables for these, or determining them programmatically? I expected the latter, but if the env var is sufficient that's easy . . .

Comment (Forgind) [reply src/Shared/CommunicationsUtilities.cs]: You mentioned they were environment variables in the issue. I can try to figure them out another way if you'd prefer. I just set them as desired on my computer.

Comment (rainersigwald) [reply src/Shared/CommunicationsUtilities.cs]: We shouldn't require user intervention to set these. I believe `dotnet build` and `dotnet msbuild` both set them for the child MSBuild process. Can you confirm that? If they both do, using the env var is probably enough.

@dsplaisted do you have an opinion on this?

Comment (dsplaisted) [reply src/Shared/CommunicationsUtilities.cs]: Isn't the path to MSBuild.dll also part of the handshake?  In that case I don't think we would additionally need to add the path to `dotnet`, as that should be less unique than the MSBuild path.

**Pull request merged**